polaris-react/src/components/Button/Button.scss

```diff
@@ -893,6 +889,10 @@
   &.iconOnly {
     margin: calc(-1 * var(--p-space-2));
 
+    #{$se23} & {
+      margin: calc(-1 * var(--p-space-1)) calc(-1 * var(--p-space-2));
+    }
+
```

Comment:
I tried a few options but this looked the best in most places in the admin. Horizontally aligned to text line heights but flush inline margins to the icon. Also note how this fixes the clipboard icon in the screenshot

|Before|After|
|-|-|
|<img width="273" alt="Screenshot 2023-06-21 at 11 09 46 PM" src="https://github.com/Shopify/polaris/assets/20652326/c4d3798f-5d42-4411-bf32-20f1222611be">|<img width="274" alt="Screenshot 2023-06-21 at 11 09 03 PM" src="https://github.com/Shopify/polaris/assets/20652326/35386e80-fbcb-4c3c-a81c-3cf584373feb">|